pkg/provisioner/ironic/testserver/server.go

```diff
@@ -115,13 +176,75 @@ func (m *MockServer) ErrorResponse(pattern string, errorCode int) *MockServer {
 func (m *MockServer) Start() *MockServer {
 	m.server = httptest.NewServer(m.mux)
 	//catch all handler
-	m.mux.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
-		m.logRequest(r, "")
-	})
+	m.mux.HandleFunc("/", m.defaultHandler)
 	return m
 }
 
 // Stop closes the server down
 func (m *MockServer) Stop() {
 	m.server.Close()
 }
+
+// AddDefaultResponseJSON adds a default response for the specified pattern
+func (m *MockServer) AddDefaultResponseJSON(patternWithVars string, httpMethod string, code int, payload interface{}) *MockServer {
+	content, err := json.Marshal(payload)
+	if err != nil {
+		m.t.Error(err)
+	}
+	return m.AddDefaultResponse(patternWithVars, httpMethod, code, string(content))
+}
+
+// AddDefaultResponse adds a default response for the specified pattern/method.
+// It is possible to use variables in the pattern using curly braces, ie `/v1/nodes/{id}/power`
+// Pattern variables can be reused in the payload, so that they will be substitued with the actual value when sending the response
+// If httpMethod is empty, the response will be applied for any method
+func (m *MockServer) AddDefaultResponse(patternWithVars string, httpMethod string, code int, payload string) *MockServer {
+
+	pattern := "^" + regexp.MustCompile("{(.[^}]*)}").ReplaceAllString(patternWithVars, "(?P<$1>.[^/]*)") + "$"
+	m.t.Logf("%s: adding default response for %s -> {%d, %s}", m.name, patternWithVars, code, payload)
```

Comment:
It would be nice to see that new pattern logged here, too, for debugging.

Reply:
Ok